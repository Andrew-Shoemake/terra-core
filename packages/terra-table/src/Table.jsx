import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './Table.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The children passed to the component
   */
  children: PropTypes.node,
  /**
   * The header passed to the table
   */
  header: PropTypes.element,
  /**
   * The padding styling to apply to the child list item content.
   * One of `'none'`, `'standard'`, `'compact'`.
   */
  paddingStyle: PropTypes.oneOf(['none', 'standard', 'compact']),
  /**
   * Function callback for the ref of the table.
   */
  refCallback: PropTypes.func,
};

const defaultProps = {
  paddingStyle: 'none',
};

const Table = ({
  children,
  header,
  paddingStyle,
  refCallback,
  ...customProps
}) => {
  const tableClassNames = cx([
    'table',
    customProps.className,
  ]);

  const attrSpread = {};
  if (paddingStyle !== 'none') {
    attrSpread['data-table-padding'] = paddingStyle;
  }

  return (
    <table {...customProps} {...attrSpread} className={tableClassNames} ref={refCallback} role="grid">
      {header}
      <tbody className={cx(['body'])}>
        {children}
      </tbody>
    </table>
  );
};

Table.propTypes = propTypes;
Table.defaultProps = defaultProps;

export default Table;
