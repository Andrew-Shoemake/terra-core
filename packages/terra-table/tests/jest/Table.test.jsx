import React from 'react';
import Table from '../../src/Table';

// Snapshot test
it('should render a default table', () => {
  const table = shallow(
    <Table />,
  );
  expect(table).toMatchSnapshot();
});

it('should render a table without zebra stripes', () => {
  const table = shallow(
    <Table isStriped={false} />,
  );
  expect(table).toMatchSnapshot();
});

it('should render a table without padding', () => {
  const table = shallow(
    <Table isPadded={false} />,
  );
  expect(table).toMatchSnapshot();
});
