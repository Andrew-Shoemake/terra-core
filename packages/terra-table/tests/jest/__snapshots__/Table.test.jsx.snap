// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default table 1`] = `
<table
  className="table"
  data-table-padded={true}
>
  <tbody
    data-table-striped={true}
  />
</table>
`;

exports[`should render a table without padding 1`] = `
<table
  className="table"
  data-table-padded={false}
>
  <tbody
    data-table-striped={true}
  />
</table>
`;

exports[`should render a table without zebra stripes 1`] = `
<table
  className="table"
  data-table-padded={true}
>
  <tbody
    data-table-striped={false}
  />
</table>
`;
